pkg/dns/azure/dns.go

```diff
@@ -132,5 +132,14 @@ func (m *provider) Replace(record *iov1.DNSRecord, zone configv1.DNSZone) error
 // getARecordName extracts the ARecord subdomain name from the full domain string.
 // azure defines the ARecord Name as the subdomain name only.
 func getARecordName(recordDomain string, zoneName string) (string, error) {
-	return strings.TrimSuffix(strings.TrimSuffix(recordDomain, "."), "."+zoneName), nil
+	trimmed := strings.TrimSuffix(strings.TrimSuffix(recordDomain, "."), "."+zoneName)
```

Comment:
This looks like it might have been okay for someone to specify a subdomain ending with a trailing ".", and not necessarily ending in the zoneName.  Do we still need to honor that API contract?

Reply:
I think the original intention of this code was to trim any trailing `.`s off of a full domain name. I don't believe we support creating an ingress controller with `spec.domain` to set to a subdomain.

Reply:
Note that this PR mirrors the current functionality of the ingress-operator's DNS management on GCP and AWS.

Reply:
Theoretically though, could someone have an ingresscontroller with `spec.domain: <domain>` for which the operator publishes a record for `<domain>.<zone>`?  And then, with this change, if the user tried to delete the ingresscontroller, deletion would fail, right?

Reply:
Ah, yea, I didn't think about the deletion flow. To mitigate this, would it make sense to have an alternative version of `getARecordName` to handle deletions safely?

Reply:
I've updated this PR to take deletion safety into account. PTAL @Miciah @candita